crates/conjure_rules/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! ### A decentralised rule registry for Conjure Oxide
+//!
+//! This crate allows registering valid functions as expression-reduction rules.
+//! Functions can be decorated with the `register_rule` macro in any downstream crate and be used by Conjure Oxide's rule engine.
+//! To achieve compile-time linking, we make use of the [`linkme`](https://docs.rs/linkme/latest/linkme/) crate.
+//!
```

Comment:
(Pedantic): This is implementation detail so should not be in the user-facing doc strings.